### PR TITLE
feature/time-limitation-for-EUD-kudos

### PR DESCRIPTION
**Pull request by rniedziolka**
Hi,
the next feature is ready:
https://www.loom.com/share/1f39f64b927546d8807a2693d813f074
Waiting for the feedback!
Thanks!

### CODE REVIEW

File: app/controllers/kudos_controller.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@ def new
   end
 
   def edit
+    authorize kudo
     if kudo.employee == current_employee
```
Comment (ciejj) [inline app/controllers/kudos_controller.rb]: [suggestion]
isn't this an authorization? :) Consider moving it into the pundit policies

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,4 +1,17 @@
...
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  def pundit_user
+    current_employee
+  end
+
+  private
+
+  def user_not_authorized
+    redirect_to kudos_path, notice: 'You can no longer edit or delete this kudo.'
```
Comment (ciejj) [inline app/controllers/application_controller.rb]: For now it is ok, but what happens if you will have more complex validations in pudit/ more reasons to fail

File: app/views/kudos/index.html.erb [hunk 3]
```diff
@@ -11,7 +11,7 @@
             <h6 class="card-subtitle mb-2 text-muted">Value: <%= kudo.company_value.title %></h6>
             <td><%= link_to 'Show', kudo, class:"btn btn-sm btn-outline-secondary" %></td>
         
-            <% if kudo.employee == current_employee %>
+            <% if policy(kudo).update? %>
```
Comment (ciejj) [inline app/views/kudos/index.html.erb]: I think without it you can see the edit buttons for other users' kudos
```suggestion
            <% if policy(kudo).update? && kudo.employee == current_employee %>
```

File: app/views/kudos/show.html.erb [hunk 4]
```diff
@@ -23,5 +23,9 @@
...
 </p>
 
-<%= link_to 'Edit', edit_kudo_path(@kudo), class:"btn btn-sm btn-outline-secondary" %>
-<%= link_to 'Back', kudos_path, class:"btn btn-sm btn-outline-secondary" %>
+<% if policy(@kudo).update? %>
+      <%= link_to 'Edit', edit_kudo_path(@kudo), class:"btn btn-sm btn-outline-secondary" %>
+      <%= link_to 'Back', kudos_path, class:"btn btn-sm btn-outline-secondary" %>
+<% else %>
+      <%= link_to 'Back', kudos_path, class:"btn btn-sm btn-outline-secondary" %>
+<% end %>
```
Comment (ciejj) [inline app/views/kudos/show.html.erb]:
```suggestion
<% if policy(@kudo).update? %>
  <%= link_to 'Edit', edit_kudo_path(@kudo), class:"btn btn-sm btn-outline-secondary" %>
<% end %>
<%= link_to 'Back', kudos_path, class:"btn btn-sm btn-outline-secondary" %>
```

**Review by ciejj — APPROVED** [review]
Good job - I have left some comments for slight improvements :) Keep going

File: app/controllers/kudos_controller.rb (hunk 1 above)
Comment (oskarlakner) [reply app/controllers/kudos_controller.rb]: this should be in the policy :)

File: app/views/kudos/show.html.erb [hunk 5]
```diff
@@ -23,5 +23,7 @@
   <%= kudo.company_value.title %>
 </p>
 
-<%= link_to 'Edit', edit_kudo_path(@kudo), class:"btn btn-sm btn-outline-secondary" %>
-<%= link_to 'Back', kudos_path, class:"btn btn-sm btn-outline-secondary" %>
+<% if policy(@kudo).update? %>
+      <%= link_to 'Edit', edit_kudo_path(@kudo), class:"btn btn-sm btn-outline-secondary" %>
```
Comment (oskarlakner) [inline app/views/kudos/show.html.erb]: the formatting is odd

**Review by oskarlakner — APPROVED** [review]

File: app/views/kudos/index.html.erb (hunk 3 above)
Comment (rniedziolka) [reply app/views/kudos/index.html.erb]: Fixed.

File: app/views/kudos/show.html.erb (hunk 4 above)
Comment (rniedziolka) [reply app/views/kudos/show.html.erb]: Fixed.

File: app/views/kudos/show.html.erb (hunk 5 above)
Comment (rniedziolka) [reply app/views/kudos/show.html.erb]: Fixed.

File: app/controllers/kudos_controller.rb (hunk 1 above)
Comment (rniedziolka) [reply app/controllers/kudos_controller.rb]: Fixed.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (rniedziolka) [reply app/controllers/application_controller.rb]: Fixed.

**Pull request merged**